src/app/App.js

```diff
@@ -287,6 +287,10 @@ class App extends EventEmitter {
 			console.log('Link clicked outside app\'s base path');
 			return false;
 		}
+		// Prevents navigation if it's a hash change on the same url.
+		if (uri.getHash() && utils.isCurrentBrowserPath(path)) {
+			return false;
+		}
```

Comment:
Hey @jbalsas, it actually changes some behavior. `canNavigate(url)` is called on the DOM event that may trigger a navigation. So, clicking or submitting a form to the same url of the browser current url is not allowed. But, if you programmatically call `.navigate()`, now it's allowed.

This is because when navigating from the history, the `popstate` event changes the url before we actually try to navigate with senna, so it's too late to check if the url we want to navigate to is the same as the one we are at.